test/instance_unittest.cc

```diff
@@ -0,0 +1,77 @@
+#include "../src/configuration.h"
+#include "../src/environment.h"
+#include "../src/instance.h"
+#include "../src/resource.h"
+#include "../src/updater.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+namespace {
+
+class InstanceTest : public ::testing::Test {
+ protected:
+  const MetadataStore::Metadata&& GetResourceMetadata(
+      const MetadataUpdater::ResourceMetadata* rm) {
+    return std::move(rm->metadata);
+  }
+
+  MonitoredResource GetMonitoredResource(
+      const MetadataUpdater::ResourceMetadata* rm) {
+    return rm->resource;
+  }
+
+  std::vector<std::string> GetResourceIds(
+      const MetadataUpdater::ResourceMetadata* rm) {
+    return rm->ids;
+  }
+};
+
+TEST_F(InstanceTest, GetInstanceMonitoredResource) {
+  Configuration config(std::istringstream(
+      "InstanceResourceType: gce_instance\n"
+      "InstanceId: 1234567891011\n"
+      "InstanceZone: us-east1-b\n"
+  ));
+  Environment env(config);
+  MonitoredResource mr_actual =
+      InstanceReader::InstanceResource(env);
+  MonitoredResource mr_expected(
```

Comment:
You can just call this `mr` and inline the call to `InstanceReader::InstanceResource(env)` into the `EXPECT_EQ`. Also below.

Reply:
You might as well inline the call being tested, then, and get rid of the local.

Reply:
Done.